crates/bindings-csharp/Runtime/Internal/FFI.cs

```diff
@@ -145,7 +147,11 @@ out RowIter out_
     );
 
     [LibraryImport(StdbNamespace)]
-    public static partial CheckedStatus _insert(TableId table_id, byte[] row, uint row_len);
+    public static partial CheckedStatus _datastore_insert_bsatn(
+        TableId table_id,
+        [MarshalUsing(CountElementName = nameof(row_len))] [Out] byte[] row,
```

Comment:
This array is used both as an input and output.

```suggestion
        [MarshalUsing(CountElementName = nameof(row_len))] [In, Out] byte[] row,
```

Reply:
Alternatively, let's maybe start using `Span` / `ReadOnlySpan` (same as Rust's `&mut [T]` / `&[T]`) when adding the new APIs.

They don't need all those extra attributes, and generally will be more flexible than arrays in terms of what they can accept as input.

We talked about switching to spans before, but it was never important enough to work on it as a refactor, but since you're adding new methods, we might as well use the better types :)